### ISSUE TITLE
iOS: add support for key commands

### ISSUE THREAD

**Issue by Dejal** [samuelclay/NewsBlur#826]
iOS 8 & 9 can access functions and navigate via key commands using an attached hardware keyboard, and iOS 9 can display a bezel to list those shortcuts when the Cmd key is held down.

That would be very useful to support.  Imagine using arrow keys to navigate around the feeds and stories, Cmd-F to go to the Find field, Cmd-N to add a new feed, and more.

**Comment by samuelclay** [comment]
Yep, @nriley brought this up yesterday. Let him work on it, he's happy to contribute.

**Comment by nriley** [comment]
Making some progress.  The biggest issues I've had thus far are:

(a) somehow the view controller isn't getting keyboard focus and I have to tap on the story before I can use the keyboard commands (right now I have it attached to StoryPageControl).

(b) I'm not sure if I'm triggering *all* of the read/unread machinery correctly as it's distributed among a 
number of classes and is pretty hard to follow.  I can either try to dig some more myself, or I can put my code up for review and see if you have any ideas about how I can rewrite it so it's clearer what's going on.

![commands](https://cloud.githubusercontent.com/assets/68303/11319721/5fa0987e-9035-11e5-8623-0b85d872bb28.png)

**Comment by nriley** [comment]
- I fixed the keyboard focus issue (had forgotten about the existence of UIResponder).  There may be some places where the responder isn't getting set correctly now, but at least this is a start.

- I realize it seems the terminology is inconsistent between the iOS and Web app.  iOS calls the RSS-feed-text "Story" (in both code and the UI) whereas the Web calls it "Feed"; there is no version of the Web's "Story" on iOS.  Confusing but if nobody's complained, maybe nobody cares? :-)

Anyway, think this is good enough for a first shot; please report bugs!

**Comment by nriley** [comment]
Things seem pretty reliable now and I can do most of my newsreading entirely from the keyboard.  (Note that I read almost exclusively from "All Stories").  I've seen some visual artifacts and multiple copies of the same story, but I don't *think* my code is responsible, although I still don't entirely understand the changes I needed to make to unread handling.  There definitely seem to be some races where the unread indicator in the story list and in the header are inconsistent, but again…

**Comment by nriley** [comment]
I'm now covering essentially anything I'd ever do with NewsBlur, so I'm going to stop now :-) If nothing else it makes navigating in the simulator a lot faster! I can squash my commits into a single one if you'd prefer.

**Comment by nriley** [comment]
Now committed to the 5.1 branch.  I'm not aware of any remaining issues, but please feel free to report.
